src/ajax/check_hgvs_dialogue.php

```diff
@@ -0,0 +1,603 @@
+
+    // Save the ['data']['DNA'] variant to the right type of mapping to
+    //  easily add it into the right fields later.
+    if ($sType == 'VOT') {
+        // If the input type was a variant on transcript, the ['data']['DNA']
+        //  field holds information on a transcript mapping.
+        $aMappedVariant['data']['transcript_mappings'][$sReferenceSequence] = $aMappedVariant['data'];
+    } else {
+        // If the input was not a VOT, it was a genomic variant, meaning we
+        //  should add the information into the genomic mappings.
+        $aMappedVariant['data']['genomic_mappings'][$sGenomeBuildID] = $aMappedVariant['data']['DNA'];
+    }
+
+    // Returning the mapping for transcript, RNA and protein variants.
+    foreach ($aMappedVariant['data']['transcript_mappings'] as $sTranscript => $aTranscriptData) {
+        // Filling in the input fields.
+        print('
+        // Adding transcript info to the fields.        
+        var oTranscriptField = $("input").filter(function() { 
+            return $(this).data("id_ncbi") == "' . $sTranscript . '" 
+        });
+        
+        if (!oTranscriptField.prop("disabled")) {
+            oTranscriptField.val("' . $aTranscriptData['DNA'] . '").attr("class", "acc");;
+            oTranscriptField.siblings("img:first").attr({src: "gfx/check.png", title: "Validated"}).show();
+            var sBaseOfFieldName = oTranscriptField.attr("name").substring(0, oTranscriptField.attr("name").indexOf("DNA"));
+            $(\'#variantForm input[name$="\' + sBaseOfFieldName + "RNA" + \'"]\').val("' . $aTranscriptData['RNA'] . '").attr("class", "acc");
+            $(\'#variantForm input[name$="\' + sBaseOfFieldName + "Protein" + \'"]\').val("' . $aTranscriptData['protein'] . '").attr("class", "acc");
+        }');
+    }
+
+    // Returning the mapping for genomic variants.
+    foreach ($aMappedVariant['data']['genomic_mappings'] as $sGBID => $aMappedGenomicVariant) {
+        if (is_string($aMappedGenomicVariant)) {
+            // The variant is a string. We don't need to make any edits.
+            $sMappedGenomicVariant = $aMappedGenomicVariant;
+
+        } elseif (count($aMappedGenomicVariant) == 1) {
+            // The variant is formatted as an array, since multiple variants were
+            //  possible. However, only one variant was found. We can simply take
+            //  the first element.
+            $sMappedGenomicVariant = $aMappedGenomicVariant[0];
+
+        } else {
+            // Multiple possible variants were found. We will inform the user,
+            //  and concatenate the variants similarly to the example below:
+            // NC_1233456.1:g.1del + NC_123456.1:g.2_3del + NC_123456.1:g.4del =
+            //  NC_123456.1:g.1del^2_3del^4del.
+            update_dialogue('<br>There were multiple genomic variant predictions for build ' . $sGBID . '.');
+
+            $sMappedGenomicVariant =
```

Comment:
This doesn't return "NC_123456.1:g.1del^2_3del^4del" as documented, but "1del^2_3del^4del".